src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -363,9 +364,9 @@ public static ListBoxModel listCheckoutCredentials(@CheckForNull Item context, S
 
     ConnectionId connectionId = new ConnectionId(apiUrl, hash);
 
-    synchronized (connections) {
-      GitHubConnection record = GitHubConnection.lookup(connectionId);
-      if (record == null) {
+    GitHubConnection record = GitHubConnection.lookup(connectionId);
+    if (record == null) {
+      synchronized (connections) {
```

Comment:
This leaves space for two threads to both think they need to construct a connection and try to add it to the HashMap.  This might be okay (not sure), except that they potentially have the same path for their `Cache` and that is forbidden.

Reply:
We probably want to use [computeIfAbsent()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-) maybe?

Reply:
Right yes let’s do that, I can adjust it tomorrow

Reply:
I'll switch to this PR and update.